compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -116,13 +116,36 @@ struct PerFunctionContext<'f> {
 
     /// Track a value's last load across all blocks.
     /// If a value is not used in anymore loads we can remove the last store to that value.
-    last_loads: HashMap<ValueId, (InstructionId, BasicBlockId)>,
+    last_loads: HashMap<ValueId, (InstructionId, BasicBlockId, u32)>,
```

Comment:
Can you add a comment explaining what the `u32` is for

Reply:
I made a context object `PerFuncLastLoadContext` and explain the fields in the struct def.